saleor/graphql/checkout/utils.py

```diff
@@ -13,3 +16,15 @@ def prepare_insufficient_stock_checkout_validation_error(exc):
         code=exc.code.value,
         params={"variants": variant_ids},
     )
+
+
+def prevent_sync_event_circular_query(func):
```

Comment:
We could add some docstring that would explain briefly the problem we are solving here i.e. why it's not allowed to resolve it in sync events.